.github/workflows/pr_push_workflow.yml

```diff
@@ -0,0 +1,62 @@
+name: PR/push
+
+on: [push, pull_request]
+
+concurrency: 
+    group: ${{ github.head_ref }}-${{ github.workflow }}
+    cancel-in-progress: true
+
+jobs:
+    Fast_build:
+        name: Fast build
+        strategy:
+            matrix:
+                os: [ubuntu-latest, windows-latest]
+                include:
+                    - os: ubuntu-latest
+                      nproc: $(nproc)
+                      pool_scalable: 'ON'
+                      jemalloc: 'ON'
+                    - os: windows-latest
+                      nproc: $Env:NUMBER_OF_PROCESSORS
+                      pool_scalable: 'OFF'
+                      jemalloc: 'OFF'
+        runs-on: ${{matrix.os}}
+
+        steps:
+        - name: Checkout repository
+          uses: actions/checkout@v4
+
+        - name: Install apt packages
+          if: matrix.os == 'ubuntu-latest'
+          run: |
+            sudo apt-get update
+            sudo apt-get install -y cmake clang libnuma-dev libjemalloc-dev libtbb-dev
+
+        - name: Configure CMake
+          run: >
+            cmake
+            -B ${{github.workspace}}/build
+            -DCMAKE_BUILD_TYPE=Release
+            -DUMF_FORMAT_CODE_STYLE=OFF
```

Comment:
shouldn't we check code style in this first check?

Reply:
It should be done in separate job with: 
```bash
$ make clang-format-check
```

Reply:
hmm.. ok, I'm not sure about this one, but I guess it may be not worth having a separate job. The build is already prepared.
Perhaps it should be just a next step in this job - as a part of preliminary check if this PR is worth "testing the full suite"

Reply:
I can prepare such a change as another PR, but currently, it does the job. Please add issue for that.

Reply:
IMHO, code style check should be a part of this (newly added in this PR) "FastBuild" job.

I'm not the decider, though, if you want to merge it like this, go ahead :) In general, this change looks good

Reply:
Added: https://github.com/oneapi-src/unified-memory-framework/issues/133
If it's okay, please resolve the discussion.